### PR TITLE
Add timezone support

### PR DESCRIPTION
**Pull request by brylie**
This pull request is the result of two other contributors' work:
-  Elliott Spira (@em0ney) - [thesaucecode:meteor-synced-cron](https://github.com/thesaucecode/meteor-synced-cron/)
- Trever Faden (@trever) - [trever:meteor-synced-cron](https://github.com/trever/meteor-synced-cron)

Timezone support may be important for projects that have users in multiple timezones, projects where the server is hosted in a different timezone than the primary users, or any other reason that UTC and the server local timezone are not sufficient.
# Todo
- [ ] Add tests
- [ ] Remove console.logs
- [ ] It's increasingly considered bad practice to redefine 'this'. The code should pass context in as an additional argument.
- [x] Document the `context` option
- [x] Don't reformat syntax for existing code
- [x] Refactor so as not to modify Later.date object. Add a new object to contain methods.
- [x] Remove the utc option 
- [x] bump the major version number.

### CODE REVIEW

**Comment by zol** [comment]
Hey, Thanks for submitting this PR. I can see how this would be a useful feature and I would be happy to merge a cleaned up PR. Here are my results from doing a short code review:

* Code lacks tests.
* Includes extraneous console.logs
* It's increasingly considered bad practice to redefine 'this'. The code should pass context in as an additional argument.
* The context option should be documented.
* Includes syntax reformatting for existing code.
* Modifies Later.date object, I would prefer if the code used a new object to avoid confusion between functionality that comes with Later vs SyncedCron.
* Should remove the utc option altogether as it's never actually used and bump the major version number.

**Comment by brylie** [comment]
OK, thanks for the review and feedback. I will work on some of the tasks mentioned above. For starters, where do you recommend I put the timezone related functions, `SyncedCron.date.timezone.example()`?

**Comment by brylie** [comment]
@em0ney and @trever would either or both of you care to help me prepare this code so that it can be merged into the main percolate branch?

**Comment by brylie** [comment]
@zol what console.logs do you want us to remove?

File: README.md [hunk 1]
```diff
@@ -17,12 +17,31 @@ To write a cron job, give it a unique name, a schedule and a function to run lik
...
+  // Set an optional context to pass in to schedule and job functions
+  context: {
+    // Set up a data context
+    // with one or more key:value pairs.
+    //  E.g.
+    userId: 'fad94af3q4ho65h4'
+  },
   schedule: function(parser) {
+    // Context object will be available as 'this'
+    console.log(this.userId);
```
Comment (zol) [inline README.md]: Remove this.

File: README.md [hunk 2]
```diff
@@ -17,12 +17,31 @@ To write a cron job, give it a unique name, a schedule and a function to run lik
...
+
     // parser is a later.parse object
     return parser.text('every 2 hours');
   },
   job: function() {
+    // Context object will be available as 'this'
+    console.log(this.userId);
```
Comment (zol) [inline README.md]: Remove this.

**Comment by zol** [comment]
`SyncedCron.date.timezone.example()` is fine.

**Comment by brylie** [comment]
@zol, the console.logs in README are used to document the context object.

**Comment by em0ney** [comment]
Hey @brylie - sorry, have been busy. What can I do to help?

**Comment by brylie** [comment]
@zol can you help with the task 'passing data context as argument'?

**Comment by brylie** [comment]
@em0ney one of the last hurdles to getting this merged is to resolve the following request from @zol:

> It's increasingly considered bad practice to redefine 'this'. The code should pass context in as an additional argument.

**Comment by brylie** [comment]
@em0ney will you please help refactor the code to pass context in as an argument? What are your thoughts on this idea?

**Comment by rudza** [comment]
Any progress on this?

**Comment by brylie** [comment]
@rudza I have made as many of the requested changes that I have capacity for. The remaining requests may need support from the primary maintainer.

**Comment by em0ney** [comment]
Hi @brylie 

Sorry for the lag on this - I'll take a look today and see what I can do.

Thanks,
Elliott

**Comment by em0ney** [comment]
Right - have just taken a look.  Can you be specific about where `this` is being redefined in the code - with line numbers?  Can't see any such example currently

**Comment by brylie** [comment]
@zol and/or would you suggest where the context should be passed in as an argument? Just for clarification. In-line comments would be appreciated.

**Comment by enzoglauber** [comment]
Any progress on this?

**Comment by brylie** [comment]
This PR was blocked by @zol, although IMO it sufficient to support the timezone feature. I was waiting on approval, and am not able to continue working on this PR.

**Comment by enzoglauber** [comment]
My biggest regret was having used this 'framework'

**Pull request closed without merging**